docs/ROADMAP.md

```diff
@@ -1,404 +1,111 @@
+1. Documentation
+2. `hyper-util`
+3. HTTP/3
+4. Observability
 
-### You call hyper, or hyper calls you?
+Each area benefits from having a top level description and goal, a place to
+track progress, and a champion (or two) that helps push the effort.
 
-> Note: this problem space, of who calls whom, will be explored more deeply in
-> a future article.
+### Documentation
 
-At times, it's been wondered whether hyper should call user code, or if user
-code should call hyper. For instance, should a `Service` be called with a
-request when the connection receives one, or should the user always poll for
-the next request.
+hyper has stabilized, so investing in documentation is wise! The way it is used
+won't change much, so documentation won't become outdated quickly. A tool that
+people don't know how to use isn't helpful at all. This helps hyper be
+**Understandable**.
 
-There's a similar question around sending a message body. Should hyper ask the
-body for more data to write, or should the user call a `write` method directly?
+The documentation focus area includes several different forms:
 
-These both get at a root topic about [write
-observability](https://github.com/hyperium/hyper/issues/2181). How do you know
-when a response, or when body data, has been written successfully? This is
-desirable for metrics, or for triggering other side-effects. 
-
-The `Service` trait also has some other frequently mentioned issues. Does
-`poll_ready` pull its complexity weight for servers? What about returning
-errors, what does that mean? Ideally users would turn all errors into
-appropriate `http::Response`s. But in HTTP/2 and beyond, stream errors are
-different from HTTP Server Error responses. Could the `Service::Error` type do
-more to encourage best practices?
-
-## Design
+- The API docs as a reference.
+- Examples as form of how-to.
+- Website guides as tutorials.
 
-The goal is to get hyper closer to the [VISION][], using that to determine the
-best way to solve the known issues above. The main thrust of the proposed
-changes are to make hyper more **Flexible** and stable.
+Each of these could benefit from dedicated planning of their overall structure,
+editing the content that already exists, and creating the rest that is sorely
+missing.
 
-In order to keep hyper **Understandable**, however, the proposed changes *must*
-be accompanied by providing utilities that solve the common usage patterns,
-documentation explaining how to use the more flexible pieces, and guides on how
-to reach for the `hyper-util`ity belt.
+### hyper-util
 
-The majority of the changes are smaller and can be contained to the *Public
-API* section, since they usually only apply to a single module or type. But the
-biggest changes are explained in detail here.
+`hyper-util` serves two main purposes:
 
-### Split per HTTP version
+1. Provide useful patterns that build on top of hyper.
+2. Explore, stabilize, and graduate some of those patterns into hyper itself.
```

Comment:
Not something we need to define now but would be interesting to write up what graduate means here and the criteria for that.